plugins/github.js

```diff
@@ -83,6 +83,58 @@ function pullRequestClosed(options, request, reply) {
 }
 
 /**
+ * Stop any running builds and start the build for the synchronized pull-request
+ * @method pullRequestClosed
+ * @param  {Object}       options
+ * @param  {String}       options.eventId    Unique ID for this GitHub event
+ * @param  {String}       options.jobId      Identifier for the Job
+ * @param  {String}       options.name       Name of the job (PR-1)
+ * @param  {Hapi.request} request Request from user
+ * @param  {Hapi.reply}   reply   Reply to user
+ */
+function pullRequestSync(options, request, reply) {
+    const eventId = options.eventId;
+    const jobId = options.jobId;
+    const name = options.name;
+
+    async.waterfall([
+        // Lookup all the builds for the jobId
+        async.apply(Build.getBuildsByJobId.bind(Build), { jobId }),
+        // Stop all builds for the jobId
+        (builds, next) => {
+            async.each(
+                builds,
+                (build, cb) => {
+                    // TODO: add in stopping of job
+                    cb();
+                },
+                next
+            );
+        },
+        // Log it
+        (next) => {
+            request.log(['webhook-github', eventId, jobId], `${name} stopped`);
+            next();
+        },
+        // Create new build for the jobId
+        async.apply(Build.create.bind(Build), { jobId }),
```

Comment:
It's because of the `async.waterfall` use and I don't like them. There should be better approaches in sequential actions and information lookup that can be done outside of a waterfall. 

At this point I'm open to suggestions